Libplanet.Net/UpdatePath.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using System.Linq;
+using Libplanet.Action;
+using Libplanet.Blockchain;
+using Libplanet.Blocks;
+
+namespace Libplanet.Net
+{
+    /// <summary>
+    /// <para>
+    /// A class representing a path from an old <see cref="BlockChain{T}.Tip"/>
+    /// to a new <see cref="BlockChain{T}.Tip"/> when a change is made to
+    /// a <see cref="BlockChain{T}"/>.
+    /// </para>
+    /// <para>
+    /// The following properties must always hold:
+    /// <list type="bullet">
+    ///     <item><description>
+    ///         <see cref="NewTip"/> always has a higher <see cref="Block{T}.TotalDifficulty"/>
+    ///         than that of <see cref="OldTip"/>.
+    ///     </description></item>
+    ///     <item><description>
+    ///         Each <see cref="Block{T}"/> in <see cref="Blocks"/> is unique.
+    ///     </description></item>
+    ///     <item><description>
+    ///         Each adjacent pair of <see cref="Block{T}"/>s in <see cref="Blocks"/> has
+    ///         <see cref="Block{T}.Index"/> difference of exactly one.
+    ///     </description></item>
+    /// </list>
+    /// </para>
+    /// </summary>
+    /// <remarks>
+    /// By the first property, <see cref="Blocks"/> should always have at least
+    /// two <see cref="Block{T}"/>s.
+    /// </remarks>
+    /// <typeparam name="T">An <see cref="IAction"/> type.  It should match
+    /// to <see cref="Block{T}"/>'s type parameter.</typeparam>
+    public class UpdatePath<T>
+        where T : IAction, new()
+    {
+        public UpdatePath(IEnumerable<Block<T>> blocks, Block<T> oldTip)
+        {
+            Blocks = blocks;
+            OldTip = oldTip;
+        }
```

Comment:
This particular implementation does not match the original specification in the documentation.
Either the documentation or the implementation should be changed to match each other.

Either way, there should be some additional sanity checks for this constructor.

Reply:
https://github.com/planetarium/libplanet/blob/3da5ba314e5c368e77b795433b4fb7ebbc1425ba/Libplanet.Net/CandidatePath.cs#L8-L36

The following properties, `NewTip`, and `Blocks` that are unique and in order, will be checked by creating `UpdatePath<T>` with `CandidateBranch<T>` that is appended successfully.

https://github.com/planetarium/libplanet/blob/17f4e97415ebeac341b9c623a18ad0b508fb5be1/Libplanet.Net/UpdatePath.cs#L9-L28

Reply:
I don't have any good idea for this... 

By using Distinct(), for checking uniqueness in `Blocks` costs `O(N)`. However, The properties are held by creating `UpdatePath<T>` with the appended `CandidateBranch<T>` to current `BlockChain<T>`, which means, they are in order, and unique.